store/mockstore/mocktikv/executor.go

```diff
@@ -396,7 +396,7 @@ func (e *indexScanExec) decodeIndexKV(pair Pair) ([][]byte, error) {
 		} else {
 			handleDatum = types.NewIntDatum(handle)
 		}
-		handleBytes, err := codec.EncodeValue(nil, b, handleDatum)
```

Comment:
Since `len(b)` is 0 here, why not reuse it?

Reply:
I also wonder this

Reply:
I also wonder this

Reply:
I forgot why I changed it... Seems the reason is that here `b` is not actually a buffer for reuse but sliced from `pair.Key`, so append to `b` may cause `handleBytes` and `pair.Key` share the same internal array. From the context, we don't know how `pair.Key` will be used, if somehow we append some bytes to `pair.Key`, it could update `handleBytes` at the same time.

Reply:
Seems to make sense.
Can we add a test case for this?

Reply:
No.